src/BSeries.jl

```diff
@@ -1897,32 +1910,45 @@ function _is_energy_preserving(trees, coefficients)
               Rational{Int8}, Rational{Int16}, Rational{Int32}, Rational{Int64},
               Rational{Int128}})
         # These types support efficient computations in sparse matrices
-        _is_energy_preserving_sparse(trees, coefficients)
+        _is_energy_preserving_sparse(trees, coefficients, atol, rtol)
     else
-        _is_energy_preserving_dense(trees, coefficients)
+        _is_energy_preserving_dense(trees, coefficients, atol, rtol)
     end
 end
 
-function _is_energy_preserving_dense(trees, coefficients)
+function _is_energy_preserving_dense(trees, coefficients, atol, rtol)
     # For every tree, obtain the adjoint and check if it exists
     length_coeff = length(trees)
     # Provided that a tree can have several adjoints, for every tree `t`
     # we need to check if there is a linear combination of the coefficients
     # of its adjoints such that this linear combination is equal to the
     # coefficient of `t`.
     # For this purpose, we create a energy_preserving_basis to store all the
-    # information
+    # information. We also include the tolerance:
     energy_preserving_basis = empty(trees)
     for (t_index, t) in enumerate(trees)
         # We do not need to consider the empty tree
         isempty(t) && continue
-
         # Check whether the level sequence corresponds to a bushy tree.
         # If so, we can exit early since the coefficients of bushy trees
         # must be zero in the modified equation of energy-preserving methods.
         if length(t) > 1 && is_bushy(t)
-            if !iszero(coefficients[t_index])
-                return false
+            if atol == 0 && rtol == 0
+                if !isapprox(coefficients[t_index],0)
```

Comment:
`isapprox(SOMETHING, 0)` will always fail - unless the first argument is exactly zero. For example,
```julia
julia> isapprox(1.0e-300, 0)
false
```

Thus, using `isapprox` here is confusing, I think

Reply:
I think it works well for `int` inputs, but for `Float` it is necessary to set a Tolerance using the keywords.

Reply:
What do you mean by

> I think it works well for `int` inputs

Could you please show me an example?

Reply:
For example, the `modified_equation` of a B-Series with integer inputs will have some coefficients `0//1`. In this case, `isapprox()` will accept it without specifying a tolerance.
`julia> isapprox(0, 0//1)
true`

Reply:
Yes - since `0//1` is exactly zero. That can already be checked by `iszero` or `== 0`. `isapprox` is usually meant to check equality with some tolerance. However, that does not work with the default tolerances when checking whether something is zero.

Reply:
Yes, I thought that we wanted the user to insert a tolerance when working with Floating inputs. Do you want to set the default tolerance to something like `1e-14`?

Reply:
I think it would be more appropriate to use something based on the precision of the floating point numbers involved as mentioned above in https://github.com/ranocha/BSeries.jl/pull/156#discussion_r1315568089